packages/edit-site/src/components/save-button/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { useEffect, useState, useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { EntitiesSavedStates } from '@wordpress/editor';
+
+export default function SaveButton() {
+	const [ , setStatus ] = useEntityProp( 'postType', 'wp_template', 'status' );
+	// Publish template if not done yet.
+	useEffect( () => setStatus( 'publish' ), [] );
+
+	const { isDirty, isSaving } = useSelect( ( select ) => {
+		const { getEntityRecordChangesByRecord, isSavingEntityRecord } = select(
+			'core'
+		);
+		const entityRecordChangesByRecord = getEntityRecordChangesByRecord();
+		const changedKinds = Object.keys( entityRecordChangesByRecord );
+		return {
+			isDirty: changedKinds.length > 0,
+			isSaving: changedKinds.some( ( changedKind ) =>
+				Object.keys(
+					entityRecordChangesByRecord[ changedKind ]
+				).some( ( changedName ) =>
+					Object.keys(
+						entityRecordChangesByRecord[ changedKind ][ changedName ]
+					).some( ( changedKey ) =>
+						isSavingEntityRecord( changedKind, changedName, changedKey )
+					)
+				)
+			),
+		};
+	} );
+	const disabled = ! isDirty || isSaving;
+
+	const [ isOpen, setIsOpen ] = useState( false );
+	const open = useCallback( setIsOpen.bind( null, true ), [] );
+	const close = useCallback( setIsOpen.bind( null, false ), [] );
+	return (
+		<>
+			<Button
+				isPrimary
+				aria-disabled={ disabled }
+				disabled={ disabled }
+				isBusy={ isSaving }
+				onClick={ disabled ? undefined : open }
+			>
+				{ __( 'Update' ) }
+			</Button>
+			<EntitiesSavedStates isOpen={ isOpen } onRequestClose={ close } />
+		</>
+	);
+}
```

Comment:
I don't like that `EntitiesSavedStates` handle the saving and that this component retrivees the status. It feels like related concerns are separate and most importantly in two separate packages. Not sure how to improve the situation though.

Reply:
Maybe `EntitiesSavedStates` should render the button?

Reply:
Or what do you think of this API?

```js
export default function SaveButton() {
	return (
		<EntitiesSavedStates>
			{ ( { isDirty, isSaving, open } ) => {
				const disabled = ! isDirty || isSaving;
				return (
					<Button
						isPrimary
						aria-disabled={ disabled }
						disabled={ disabled }
						isBusy={ isSaving }
						onClick={ disabled ? undefined : open }
					>
						{ __( 'Update Design' ) }
					</Button>
				);
			} }
		</EntitiesSavedStates>
	);
}
```

Reply:
It does seem reasonable but I know we've been talking about rethinking the saving flow (pre/post publish...) regarding FSE so it could have an impact on the API. Maybe we can start as it is right now but not push too far this direction api wise until we're certain of the flows.